cpp/ps0004/ps0004_soln.cpp

```diff
@@ -4,20 +4,29 @@ using namespace std;
 
 int main()
 {
- int a = 65;
+    // prompting user for number 1/1.5
+    float num;
+    cin >> num;
 
-cin >> a;
-    char character1 = (char)a;
-
-    if (a>=32 && a<=126)
+    // checking invalidaty of user's input 2/4
+	   // printing the number and the opposite printable character 2/2
+    if (num == 0)
+    {
+       cout << "Invalid character/s or zero value!" << endl;
+    }
+    else if (num != (int) num) {
+       cout << "Invalid: a floating point number!" << endl;
+    }
+    else if (num>=32 && num<=126)
+    {
+       char character = (char)num;
+       cout << "The character for ASCII value " << num << " is " << character << endl;
+    }
+    else
     {
-       cout << "The character for ASCII value " << a
-         << " is " << character1 << endl;
+       cout << "Invalid: number is out of range!";
     }

```

Comment:
- [x] used `if` structure in a right way
- [ ] `if` structure take in consideration all invalid cases
- [ ] all expressions to check invalidity are correct
- [ ] used `std::cout` and `std::endl` or `\n` to print a different message for the 3 cases
- [x] converted the number to the opposite printable character
- [x] printed the number and the opposite printable character